ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -543,4 +552,49 @@ export class AptosClient {
     const tableItem = await this.tables.getTableItem(handle, data, params);
     return tableItem;
   }
+
+  // Only cache for a short period to avoid excessive amount of memory usage
+  @MemoizeExpiring(2 * 60 * 1000) // Cache for 2min
+  async getTxnBuilderWithABI(accountFrom: AptosAccount): Promise<TransactionBuilderEd25519> {
+    if (this.abis.length === 0) {
+      throw new Error("ABIs are not provided.");
+    }
+
+    const [{ sequence_number: sequenceNumber }, chainId] = await Promise.all([
+      this.getAccount(accountFrom.address()),
+      this.getChainId(),
+    ]);
+
+    const rawTxnBuilder = new TransactionBuilderABI(this.abis, {
+      sender: accountFrom.address(),
+      sequenceNumber,
+      chainId,
+    });
+
+    return new TransactionBuilderEd25519(
+      (signingMessage: TxnBuilderTypes.SigningMessage) => {
+        const sigHexStr = accountFrom.signBuffer(signingMessage);
+        return new TxnBuilderTypes.Ed25519Signature(sigHexStr.toUint8Array());
+      },
+      accountFrom.pubKey().toUint8Array(),
+      rawTxnBuilder,
+    );
+  }
+
+  async sendABITransaction(
```

Comment:
interesting... though part of me thinks we should actually support the ability to pull the ABI down and make it so that the user can review it. Sort of like a self-expanding SDK. This idea feels a little too dynamic for me. What are your thoughts? How would a user learn about the ABI?

Reply:
Added more info about this PR. This PR uses the ABI generated by the Move compiler. This is similar to Solidity. Solidity developers compile sols and produce ABIs and then provide the ABIs to web3 js. 

Pulling the ABI down is also possible with our chain since our REST APIs support it. We actually can generate TS code for modules and script functions based on the information of https://fullnode.devnet.aptoslabs.com/accounts/0x01/modules, which I think is even better. I can work on another PR if we think that is a good idea.

Reply:
Different from EVM, given an address and module name, we can fetch and load the ABI from the chain. Can we use that here instead of having the users specify the abi? May be you're already doing this and I misunderstood things :)

Reply:
I am not doing that in this PR. This PR requires users to specify the ABI bytes. For anything else, I would rather defer to another PR. This PR is getting large and hard to understand already :)